index.js

```diff
@@ -184,6 +191,24 @@ function numEventLoopLags() {
   return result;
 }
 
+// Returns an array of the most recent BHR hang pseudo-stacks as strings
+function mostRecentHangStacks() {
+  let geckoThread = Services.telemetry.threadHangStats.find(thread =>
+    thread.name == "Gecko"
+  );
+  if (!geckoThread) {
+    console.warn("Uh oh, there doesn't seem to be a thread with name \"Gecko\"!");
+    return [];
+  }
+  if (geckoThread.hangs.length == 0) { // No hangs found
+    return [];
+  }
+  var recentHangs = geckoThread.hangs.slice(Math.max(0, geckoThread.hangs.length - 10));
+  return recentHangs.map((hangEntry, i) => {
+    return hangEntry.stack.slice(0).reverse().join("\n");
```

Comment:
not sure you need the `slice(0)` for the shallow copy since you already got a shallow copy from the `slice(-10)` in line 206